djangobaseview/static/js/guestbook/view/Greeting.js

```diff
@@ -0,0 +1,96 @@
+define([
+	"dojo/_base/declare",
+	"dojo/_base/fx",
+	"dojo/_base/lang",
+	"dojo/on",
+	"dojo/query",
+	"dojo/dom-style",
+	"dojo/dom",
+	"dojo/text!./templates/Greeting.html",
+	"dijit/registry",
+	"dijit/form/Button",
+	"dijit/form/Form",
+	"dijit/form/TextBox",
+	"./_ViewBaseMixin"
+
+], function(declare, baseFx, lang, on, query, domStyle, dom, template,
+			registry, button, form, textbox, _ViewBaseMixin){
+
+		return declare("WidgetGreeting", [_ViewBaseMixin], {
+
+			templateString: template,
+			guestbookStore: null,
+			guestbookName: "",
+			greeting_id: "",
+			author: "",
+			content: "",
+			date: "",
+
+			constructor: function(data){
+				this.guestbookName = data.guestbook_name;
+				this.greeting_id = data.greeting_id;
+				this.content = data.content;
+				this.date = data.date;
+				if (data.updated_by !== "None"){
+					this.author = data.updated_by;
+				}else{
+					this.author = "Anonymous Person";
+				}
+				this.guestbookStore = data.guestbook_store;
+			},
+
+			postCreate: function(data){
+				this.inherited(arguments);
+
+				this.own(
+					on(this.deleteButton,
+						"click", lang.hitch(this, "deleteGreeting")),
+					on(this.openEditFormButton,
+						"click", lang.hitch(this, "openEditForm")),
+					on(this.submitUpdateGreetingButton,
+						"click", lang.hitch(this, "updateGreeting"))
+				);
+			},
+
+			deleteGreeting: function(){
+				this.guestbookStore.set("guestbookName", this.guestbookName);
```

Comment:
thao tác `guestbookStore.set("guestbookName", ... ` nên đóng gói vào trong `this.guestbookStore.deleteGreeting()` để giảm thiểu sai sót khi người khác xài `guestbookStore ` do mình viết ra